cgi-bin/LJ/Support.pm

```diff
@@ -907,6 +907,40 @@ sub touch_request
     return 1;
 }
 
+# Extra email addresses are stored as support properties
+# - nb_extra_addresses: number of extra addresses (if not present, 0)
+# - extra_address_$n: extra address $n (0<=$n<nb_extra_addresses)
+
+sub add_email_address {
+    my ( $sp, $address ) = @_;
+
+    # Already present?
+    return if grep { $_ eq $address } @{all_email_addresses( $sp )};
+
+    # Add
+    my $props = load_props( $sp->{spid} + 0 );
+    my $nb_extra_addresses = $props->{nb_extra_addresses} || 0;
+    set_prop( $sp->{spid}, 'nb_extra_addresses', $nb_extra_addresses + 1 );
```

Comment:
I'm not sure that expectation is warranted in the long term, considering this comment in LJ::Support::set_prop:

> # TODO:
> # -- delete on 'undef' propval
> # -- allow setting of multiple

I also considered putting addresses in a table of their own, but that feels overwrought to me.

And I think that data safety is the overriding bit here, considering the possible legal consequences of not emailing one address you thought you had. (This is for ToS, after all.)